_posts/2025-05-05-plan-execute-agent.md

```diff
@@ -0,0 +1,447 @@
+---
+layout: post
+title: "Intelligent troubleshooting using OpenSearch 3.0's plan-execute-reflect agent"
+authors:
+  - pyek-bot
+  - ylwu
+  - kolchfa
+date: 2025-05-05
+categories:
+  - technical-posts
+meta_keywords: Plan and Execute, OpenSearch, AI, Agents, Function calling, Claude
+meta_description: Learn about OpenSearch 3.0's agent framework enhancements and utilize the plan-execute-reflect agent to resolve an observability use case.
+---
+OpenSearch 3.0 introduces the [_plan–execute–reflect agent_](https://docs.opensearch.org/docs/latest/ml-commons-plugin/agents-tools/agents/plan-execute-reflect/)---a powerful new capability that breaks down complex problems, selects and executes tools autonomously, and adapts through reflection. In this post, we'll show you how this agent automates root cause analysis in observability workflows.
```

Comment:
Link OpenSearch 3.0 to the release blog here: https://opensearch.org/blog/unveiling-opensearch-3-0/